src/backend/distributed/commands/utility_hook.c

```diff
@@ -912,6 +912,20 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 		{
 			CommitTransactionCommand();
 			StartTransactionCommand();
```

Comment:
I think the previous `CommitTransactionCommand ()` should have released the virtualtxId:
```
  * frame #0: 0x0000000101142b9d postgres`VirtualXactLockTableCleanup at lock.c:4329:2
    frame #1: 0x0000000101142674 postgres`LockReleaseAll(lockmethodid=1, allLocks=<unavailable>) at lock.c:2117:3
    frame #2: 0x00000001011462d7 postgres`ProcReleaseLocks(isCommit=<unavailable>) at proc.c:795:2
    frame #3: 0x0000000101266c0c postgres`ResourceOwnerReleaseInternal(owner=0x00007fbf838316a8, phase=RESOURCE_RELEASE_LOCKS, isCommit=<unavailable>, isTopLevel=<unavailable>) at resowner.c:568:5
    frame #4: 0x0000000100f064f9 postgres`CommitTransaction at xact.c:2217:2
```

Reply:
yeah, I couldn't explain it precisely but yes, it does that.
However, then we open a new transaction so now we have the lock on vxid for the new xact